homeassistant/components/thermostat/zwave.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+# @Author: Robert D. Cotey II <coteyr@coteyr.net>
+# @Date:   2016-02-24 17:52:01
+# @Last Modified by:   Robert D. Cotey II <coteyr@coteyr.net>
+# @Last Modified time: 2016-02-26 22:28:05
+"""
+ZWave Thermostat.
+
+"""
+DOMAIN = 'thermostat'
+
+CONF_NAME = 'name'
+DEFAULT_NAME = 'ZWave Thermostat'
+
+from homeassistant.components.thermostat import DOMAIN
+from homeassistant.components.thermostat import ThermostatDevice
+from homeassistant.const import TEMP_CELCIUS, TEMP_FAHRENHEIT
+from homeassistant.components.zwave import (
+    ATTR_NODE_ID, ATTR_VALUE_ID, COMMAND_CLASS_ALARM, COMMAND_CLASS_METER,
+    COMMAND_CLASS_SENSOR_MULTILEVEL, NETWORK, COMMAND_CLASS_THERMOSTAT_MODE,
+    TYPE_DECIMAL, ZWaveDeviceEntity)
+from homeassistant.helpers.entity import Entity
+
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the ZWave thermostats."""
+    if discovery_info is None or NETWORK is None:
+      return
+
+    node = NETWORK.nodes[discovery_info[ATTR_NODE_ID]]
+    value = node.values[discovery_info[ATTR_VALUE_ID]]
+    value.set_change_verified(False)
+   # if value.command_class == COMMAND_CLASS_THERMOSTAT_MODE:
+    add_devices([ZWaveThermostat(value)])
+
+
+# pylint: disable=too-many-arguments
+class ZWaveThermostat(ZWaveDeviceEntity, ThermostatDevice, Entity):
+    """Represents a HeatControl thermostat."""
+    def __init__(self, value):
+        from openzwave.network import ZWaveNetwork
+        from pydispatch import dispatcher
+        # ZWaveDeviceEntity.__init__(self, value, DOMAIN)
+        ZWaveDeviceEntity.__init__(self, value, DOMAIN)
+        self._current_temperature = 0
+        self._target_temperature = 71
+        self._unit_of_measurement = TEMP_FAHRENHEIT
+        self._away = False
+        self._current_temperature = 90
+        self._node = value.node
+        self._current_operation = "Idle"
+        self._current_operation_state = "Idle"
+
+        # register listener
+        dispatcher.connect(
+            self.value_changed, ZWaveNetwork.SIGNAL_VALUE_CHANGED)
+
+    def value_changed(self, value):
+      """Called when a value has changed on the network."""
+      if self._value.node == value.node:
+        self.update_properties()
+        self.update_ha_state()
+        print(value.data)
+
+    def update_properties(self):
+      for key, value in self._node.get_values_for_command_class(0x43).items(): # set point
+        if int(value.data) != 0:
+          self._target_temperature = value.data
```

Comment:
This is one I want to fix. I want to try them all. My thermostat for example responds with 0 and 70 when the heat is on and 70 and 0 when the AC is on. If I have Aux Heat on it responds with 0 70 70. I have yet to see an instance where the last non-zero integer is the different from the first, but until I try them all... That being said, it would be better not to loop at all. I just can't find a way to do that.